azure-nextgen-cs-credential-rotation-one-set/AppStack.cs

```diff
@@ -0,0 +1,304 @@
+// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.
+
+using System;
+using Pulumi;
+using Pulumi.AzureNextGen.Authorization.Latest;
+using Pulumi.AzureNextGen.EventGrid.V20200401Preview.Inputs;
+using Pulumi.AzureNextGen.EventGrid.V20200401Preview;
+using Pulumi.AzureNextGen.Insights.Latest;
+using Pulumi.AzureNextGen.KeyVault.Latest;
+using Pulumi.AzureNextGen.KeyVault.Latest.Inputs;
+using Pulumi.AzureNextGen.Resources.Latest;
+using Pulumi.AzureNextGen.Sql.Latest;
+using Pulumi.AzureNextGen.Storage.Latest;
+using Pulumi.AzureNextGen.Web.Latest;
+using Pulumi.AzureNextGen.Web.Latest.Inputs;
+using KeyVault = Pulumi.AzureNextGen.KeyVault.Latest;
+using Storage = Pulumi.AzureNextGen.Storage.Latest;
+using ManagedServiceIdentityType = Pulumi.AzureNextGen.Web.Latest.ManagedServiceIdentityType;
+
+internal class AppStack : Stack
+{
+    [Output]
+    public Output<string> WebAppEndpoint { get; set; }
+
+    public AppStack()
+    {
+        var config = new Config();
+        var resourceGroupName = config.Get("resourceGroupName") ?? "rotatesecretoneset";
+        var resourceNamePrefix = config.Get("resourceNamePrefix") ?? resourceGroupName;
+        var location = config.Get("location") ?? "EastUS";
+
+        var resourceGroup = new ResourceGroup("resourceGroup", new ResourceGroupArgs { ResourceGroupName = resourceGroupName, Location = location });
+
+        var sqlAdminLogin = config.Get("sqlAdminLogin") ?? "sqlAdmin";
+        var sqlAdminPassword = Guid.NewGuid().ToString();
+        var sqlServer = new Server("sqlServer", new ServerArgs
+        {
+            AdministratorLogin = sqlAdminLogin,
+            AdministratorLoginPassword = sqlAdminPassword,
+            Location = location,
+            ResourceGroupName = resourceGroup.Name,
+            ServerName = $"{resourceNamePrefix}-sql",
+            Version = "12.0",
+        });
+
+        new FirewallRule("firewallRule",
+            new FirewallRuleArgs
+            {
+                FirewallRuleName = "AllowAllWindowsAzureIps",
+                ServerName = sqlServer.Name,
+                ResourceGroupName = resourceGroup.Name,
+                StartIpAddress = "0.0.0.0",
+                EndIpAddress = "0.0.0.0",
+            });
+
+        var clientConfig = Output.Create(GetClientConfig.InvokeAsync());
+        var tenantId = clientConfig.Apply(c => c.TenantId);
+
+        var storageAccountName = $"st{resourceNamePrefix}";
+        var storageAccount = new StorageAccount("storageAccount", new StorageAccountArgs
+        {
+            AccountName = storageAccountName,
+            Kind = "Storage",
+            Location = location,
```

Comment:
If you update to 0.6.1, the location shouldn't be a required parameter anymore, similar to the classic Azure provider

Reply:
Done.